File: tvr-core/src/main/java/io/github/binaryfoo/decoders/apdu/CommandAPDUDecoder.java
package io.github.binaryfoo.decoders.apdu;

import io.github.binaryfoo.DecodedData;
import io.github.binaryfoo.decoders.DecodeSession;

public interface CommandAPDUDecoder {
    APDUCommand getCommand();

    DecodedData decode(String input, int startIndexInBytes, DecodeSession session);
}

File: src/main/java/io/github/binaryfoo/decoders/apdu/CommandAPDUDecoder.kt
package io.github.binaryfoo.decoders.apdu

import io.github.binaryfoo.DecodedData
import io.github.binaryfoo.decoders.DecodeSession

interface CommandAPDUDecoder {
  fun getCommand(): APDUCommand

  fun decode(input: String, startIndexInBytes: Int, session: DecodeSession): DecodedData
}
